src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -330,27 +365,36 @@ private JToken ReadJToken(ReadOnlySequence<byte> contentBuffer, Encoding encodin
         /// <param name="jsonRpcMessage">A JSON-RPC message.</param>
         private void TokenizeUserData(JsonRpcMessage jsonRpcMessage)
         {
-            if (jsonRpcMessage is Protocol.JsonRpcRequest request)
+            try
             {
-                if (request.ArgumentsList != null)
+                if (jsonRpcMessage is Protocol.JsonRpcRequest request)
                 {
-                    request.ArgumentsList = request.ArgumentsList.Select(this.TokenizeUserData).ToArray();
-                }
-                else if (request.Arguments != null)
-                {
-                    if (this.ProtocolVersion.Major < 2)
+                    this.requestId = Convert.ToInt64(request.Id);
+
+                    if (request.ArgumentsList != null)
                     {
-                        throw new NotSupportedException(Resources.ParameterObjectsNotSupportedInJsonRpc10);
+                        request.ArgumentsList = request.ArgumentsList.Select(this.TokenizeUserData).ToArray();
+                    }
+                    else if (request.Arguments != null)
+                    {
+                        if (this.ProtocolVersion.Major < 2)
+                        {
+                            throw new NotSupportedException(Resources.ParameterObjectsNotSupportedInJsonRpc10);
+                        }
+
+                        // Tokenize the user data using the user-supplied serializer.
+                        var paramsObject = JObject.FromObject(request.Arguments, this.JsonSerializer);
+                        request.Arguments = paramsObject;
                     }
-
-                    // Tokenize the user data using the user-supplied serializer.
-                    var paramsObject = JObject.FromObject(request.Arguments, this.JsonSerializer);
-                    request.Arguments = paramsObject;
+                }
+                else if (jsonRpcMessage is Protocol.JsonRpcResult result)
+                {
+                    result.Result = this.TokenizeUserData(result.Result);
                 }
             }
-            else if (jsonRpcMessage is Protocol.JsonRpcResult result)
+            finally
             {
-                result.Result = this.TokenizeUserData(result.Result);
+                this.requestId = 0;
```

Comment:
> this.requestId = 0; [](start = 15, length = 20)

Remember that 0 is a valid ID. So given your intent is to indicate that we're no longer reading a request, we should set this to `null` (which requires that `requestId` be defined as a `long?` field.